src/irmin-pack/unix/file_manager.ml

```diff
@@ -692,33 +687,42 @@ struct
             | `Unknown_major_pack_version _ ) as e ->
             e)
 
-  let swap t ~generation ~new_suffix_start_offset ~new_suffix_end_offset
-      ~latest_gc_target_offset =
+  let swap t ~generation ~suffix_start_offset ~chunk_start_idx ~chunk_num
+      ~suffix_dead_bytes ~latest_gc_target_offset =
     let open Result_syntax in
     [%log.debug
-      "Gc in main: swap %d %#d %#d" generation
-        (Int63.to_int new_suffix_start_offset)
-        (Int63.to_int new_suffix_end_offset)];
+      "Gc in main: swap gen %d; suffix start %a; chunk start idx %d; chunk num \
+       %d; suffix dead bytes %a"
+      generation Int63.pp suffix_start_offset chunk_start_idx chunk_num Int63.pp
+        suffix_dead_bytes];
     let c0 = Mtime_clock.counter () in
-    let pl = Control.payload t.control in
-    (* Opening the suffix requires passing it its length. We compute it from the
-       global offsets
-
-       TODO: this calculation of [suffix_end_poff] only works with one chunk.
-       this code will be removed once we have chunk GC.
-    *)
-    let suffix_end_poff =
-      let open Int63.Syntax in
-      new_suffix_end_offset - new_suffix_start_offset
+    let pl =
```

Comment:
this is duplicated below. I prefer to keep the control file update in step 2, as it is now, if that's ok with you

Reply:
Ah, good catch! This is duplicated because I missed it in the rebase. I previously had a reason for the new control file payload to be created before reopening, but I will fix this.